### PR TITLE
Certora Renew

### PR DESCRIPTION
**Pull request by MartinGbz**

### CODE REVIEW

File: src/20241014_AaveV3Ethereum_AaveCertoraContinuousSecurityServices/AaveV3Ethereum_AaveCertoraContinuousSecurityServices_20241014.sol [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import {MiscEthereum} from 'aave-address-book/MiscEthereum.sol';
+
+/**
+ * @title Aave <> Certora Continuous Security Services
+ * @author Aave Chan Initiative
+ * - Snapshot: https://snapshot.org/#/aave.eth/proposal/0xebf0b33be0c3784b2928112414f08e31ac57705f49d46668bfef6fa6f761141d
+ * - Discussion: https://governance.aave.com/t/arfc-aave-certora-continuous-security-services/19262
+ */
+contract AaveV3Ethereum_AaveCertoraContinuousSecurityServices_20241014 is IProposalGenericExecutor {
+  address public constant CERTORA_RECEIVER = 0xE8555F05b3f5a1F4566CD7da98c4e5F195258B65;
```
Comment (MartinGbz) [inline src/20241014_AaveV3Ethereum_AaveCertoraContinuousSecurityServices/AaveV3Ethereum_AaveCertoraContinuousSecurityServices_20241014.sol]: This is llamarisk address
Certora address is: `0x0F11640BF66e2D9352d9c41434A5C6E597c5e4c8`

File: src/20241014_AaveV3Ethereum_AaveCertoraContinuousSecurityServices/AaveV3Ethereum_AaveCertoraContinuousSecurityServices_20241014.sol [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      CollectorUtils.CreateStreamInput({
+        underlying: AaveV3EthereumAssets.GHO_UNDERLYING,
+        receiver: CERTORA_RECEIVER,
+        amount: 1_150_000 * 10 ** IERC20Metadata(AaveV3EthereumAssets.GHO_UNDERLYING).decimals(),
+        start: block.timestamp,
+        duration: STOP_TIME - block.timestamp // ends on 11 september 2025
+      })
+    );
+
+    uint256 DURATION = STOP_TIME - block.timestamp;
```
Comment (MartinGbz) [inline src/20241014_AaveV3Ethereum_AaveCertoraContinuousSecurityServices/AaveV3Ethereum_AaveCertoraContinuousSecurityServices_20241014.sol]: You can put `DURATION` at the top of `execute` function

File: src/20241014_AaveV3Ethereum_AaveCertoraContinuousSecurityServices/AaveV3Ethereum_AaveCertoraContinuousSecurityServices_20241014.sol [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    uint256 AAVE_AMOUNT = (550_000 *
+      10 ** IERC20Metadata(AaveV3EthereumAssets.AAVE_UNDERLYING).decimals() *
+      10 ** 8) / AAVE_PRICE;
+    uint256 ACTUAL_AMOUNT = (AAVE_AMOUNT / DURATION) * DURATION;
```
Comment (MartinGbz) [inline src/20241014_AaveV3Ethereum_AaveCertoraContinuousSecurityServices/AaveV3Ethereum_AaveCertoraContinuousSecurityServices_20241014.sol:38]: (AAVE_AMOUNT / DURATION) * DURATION === AAVE_AMOUNT, so why not just using AAVE_AMOUNT?

+ Why not create also a GHO_AMOUNT at the top of `execute` function?

File: src/20241014_AaveV3Ethereum_AaveCertoraContinuousSecurityServices/AaveV3Ethereum_AaveCertoraContinuousSecurityServices_20241014.sol [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        amount: 1_150_000 * 10 ** IERC20Metadata(AaveV3EthereumAssets.GHO_UNDERLYING).decimals(),
+        start: block.timestamp,
+        duration: STOP_TIME - block.timestamp // ends on 11 september 2025
+      })
+    );
+
+    uint256 DURATION = STOP_TIME - block.timestamp;
+    uint256 AAVE_AMOUNT = (550_000 *
+      10 ** IERC20Metadata(AaveV3EthereumAssets.AAVE_UNDERLYING).decimals() *
+      10 ** 8) / AAVE_PRICE;
```
Comment (MartinGbz) [inline src/20241014_AaveV3Ethereum_AaveCertoraContinuousSecurityServices/AaveV3Ethereum_AaveCertoraContinuousSecurityServices_20241014.sol]: Maybe precise it's AAVE price is 8 decimals in order to understand the `10 ** 8`

File: src/20241014_AaveV3Ethereum_AaveCertoraContinuousSecurityServices/AaveV3Ethereum_AaveCertoraContinuousSecurityServices_20241014.sol (hunk 3 above)
Comment (Rozengarden) [reply src/20241014_AaveV3Ethereum_AaveCertoraContinuousSecurityServices/AaveV3Ethereum_AaveCertoraContinuousSecurityServices_20241014.sol:38]: In solidity "/" is actualy the euclidiean division, so this is equivalent to `amount - amount % duration`

For GHO am actualy using `stream` a wrapper function that does it automagically before calling `createStream`

Comment (MartinGbz) [reply src/20241014_AaveV3Ethereum_AaveCertoraContinuousSecurityServices/AaveV3Ethereum_AaveCertoraContinuousSecurityServices_20241014.sol:38]: Ah okay my bad, I'm not used to solidity, I forgot this particularity.
And indeed for the GHO, I've just check the `CollectorUtils.sol` contract.

**Pull request closed without merging**